### ISSUE TITLE
Question about the comparison between AIRL and f-IRL

### ISSUE THREAD

**Issue by sff1019** [twni2016/f-IRL#1]
Hi Tianwei,

Thank you for this amazing work, and for making your repo publicly available.
I am reaching out to ask about the implementation of AIRL.

It seems like you have changed the AIRL algorithm to be 'off-policy by incorporating SAC into the training loop. Hence, to train the discriminator, you've also changed the training data to be sampled from the replay-buffer, which is an off-policy manner.
However, f-IRL does not mimic this, and instead, you are sampling data that will be used to train the discriminator from the current policy, which is an on-policy manner.
I believe this is not a fair comparison between the two and is one of the reasons why AIRL is not performing well.

What do you think of this?

Thank you,
Hana

**Comment by twni2016** [comment]
Hi Hana,

Thank you for your interest. 

> It seems like you have changed the AIRL algorithm to be 'off-policy by incorporating SAC into the training loop. 

Yes, this follows f-MAX training style to use off-policy data for AIRL disc. I am not quite sure if original AIRL disc can be only trained with on-policy data. Maybe you can have a try. 

> I believe this is not a fair comparison between the two and is one of the reasons why AIRL is not performing well.

By the way, there is an important update in AIRL training code, https://github.com/twni2016/f-IRL/blob/master/common/sac_agent.py#L65 there is a bug in `log_prob` function which is ONLY used in AIRL. I have fixed it, but have not tested it on the benchmarks. Maybe this is the reason why our re-implemented AIRL not works. Sorry for that.

> However, f-IRL does not mimic this, and instead, you are sampling data that will be used to train the discriminator from the current policy, which is an on-policy manner.

Yes, you are right. By the way, the "discriminator" is called the reward function in our algorithm.
